### PR TITLE
DataOutResample for multiple components

### PR DESCRIPTION
**Pull request by peterrum**
FYI @mschreter @jh66637

### CODE REVIEW

File: source/numerics/data_out_resample.cc [hunk 1]
```diff
@@ -151,32 +151,42 @@ DataOutResample<dim, patch_dim, spacedim>::build_patches(
...
+          ExcMessage(
+            "This class currently only supports scalar elements and elements "
+            "with a single base element."));
+#endif
+
+      for (unsigned int comp = 0; comp < dh.get_fe_collection().n_components();
+           ++comp)
+        {
+          const auto values = VectorTools::point_values<1>(
+            rpe, dh, data_ptr->vector, VectorTools::EvaluationFlags::avg, comp);
```
Comment (peterrum) [inline source/numerics/data_out_resample.cc:167]: This is the only relevant change. Loop over all components and call `VectorTools::point_values` for each component.

File: include/deal.II/numerics/vector_tools_evaluate.h [hunk 2]
```diff
@@ -547,8 +560,9 @@ namespace VectorTools
                                           typename VectorType::value_type>>> &)
     {
       (void)evaluation_flags;
+      (void)first_selected_component;
 
-      Assert(n_components == 1,
+      Assert(n_components == 1 && first_selected_component == 0,
```
Comment (marcfehling) [inline include/deal.II/numerics/vector_tools_evaluate.h:565]: That's the only place where you use `first_selected_component`, right? Did I overlook something?

**Review by marcfehling — APPROVED** [review]

**Comment by marcfehling** [comment]
Would you document the new parameter in the header file? I would also like to have a test if that's alright :)

File: include/deal.II/numerics/vector_tools_evaluate.h (hunk 2 above)
Comment (peterrum) [reply include/deal.II/numerics/vector_tools_evaluate.h:565]: It is used here:

https://github.com/dealii/dealii/blob/29d31b186cdbf6e0e6bc0ad8f309753cb97f6d30/include/deal.II/numerics/vector_tools_evaluate.h#L446-L453

**Comment by peterrum** [comment]
> Would you document the new parameter in the header file? I would also like to have a test if that's alright :)

@marcfehling Done!

**Review by marcfehling — APPROVED** [review]

**Pull request merged**